lib/scanner/walk.go

```diff
@@ -90,9 +88,32 @@ type CurrentFiler interface {
 	CurrentFile(name string) (protocol.FileInfo, bool)
 }
 
+type MtimeRepo interface {
+	// GetMtime returns a (possibly modified) actual mtime given a file name
+	// and it's on disk mtime.
+	GetMtime(relPath string, mtime time.Time) time.Time
+}
+
+func Walk(cfg Config) (chan protocol.FileInfo, error) {
+	if cfg.CurrentFiler == nil {
+		cfg.CurrentFiler = noCurrentFiler{}
+	}
+	if cfg.TempNamer == nil {
+		cfg.TempNamer = noTempNamer{}
+	}
+	if cfg.MtimeRepo == nil {
+		cfg.MtimeRepo = noMtimeRepo{}
+	}
+	return (&walker{cfg}).walk()
+}
```

Comment:
Can we change this to be:
```
w := &walker{cfg}

if w.TempNamer == nil {
...

return w.walk()
```

This gets rid of the braces around (&walker{cfg}) which looks unusual.